<!doctype html><html data-page="history" lang="en"><head><title>History | TogoVar</title><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><link rel="icon" href="/favicon.svg" type="image/svg+xml"><script>(async function() {
        try {
          if (window.location.origin === 'http://localhost:8000') return
          const response = await fetch(`${window.location.origin}/auth/status`);
      
          let isLogin = false;
          if (response.status === 200 || response.status === 403) {
            isLogin = true;
          }
      
          const loginMenu = document.querySelector('.menu-button.login a.link');
          if (isLogin === true) {
            loginMenu.href = "/auth/logout";
            loginMenu.textContent = "Logout";
          } else {
            loginMenu.href = "/auth/login";
            loginMenu.textContent = "Login";
          }
      
        } catch (error) {
          console.error('Error fetching auth status:', error);
        }
      })();</script><link href="/css/main-06f76ee01cc97734dbff.css" rel="stylesheet"></head><body><header id="GlobalHeader"><div class="left"><h1 class="heading"><a class="link" href="/"><img class="logotype" src="/images/logotype-9dd64508bb1747791269e1d69fc69389.svg" alt="TOGO VAR"></a></h1><span class="summary">A comprehensive Japanese genetic variation database</span><nav class="dropdown-view"><button class="button">GRCh37</button><div class="selectshow"><ul class="list"><li class="item"><a class="link -active">GRCh37</a></li><li class="item"><a class="link" href="http://grch38.togovar.org/">GRCh38</a></li></ul></div></nav></div><div class="right"><div class="menu-bar"></div><nav class="menus-view"><div class="menu-wrapper"><ul class="menu navigation"><li class="item menu-button"><a class="link" href="/">Home</a></li><li class="item menu-button"><a class="link" href="/doc/datasets">Datasets</a></li><li class="item menu-button"><a class="link" href="/doc/downloads">Downloads</a></li><li class="item menu-button"><a class="link" href="/api" target="_blank">API</a></li><li class="item menu-button"><a class="link" href="/doc/terms">Terms</a></li><li class="item menu-button"><a class="link" href="/doc/contact">Contact</a></li><li class="item menu-button"><a class="link" href="/doc/about">About</a></li><li class="item menu-button -current"><a class="link" href="/doc/history">History</a></li><li class="item menu-button"><a class="link" href="/doc/help">Help</a></li></ul><ul class="menu"><li class="menu-button languageswitcher en"><a class="language en" href="/doc/history">en</a><a class="language ja" href="/doc/ja/history">ja</a></li><li class="menu-button login"><a class="link"></a></li></ul></div></nav></div></header><main><article class="common-document"><header class="header"><h2 class="title">Update history</h2></header><div class="contents"><section class="section"><h3 class="sectiontitle">2025/6/4</h3><ul class="unordered-list"><li><strong>Gene page:</strong> <a class="hyper-text -external" href="https://molossinus.brc.riken.jp/mogplus/">MoG+ (Mouse genome database with high added value)</a>was newly added.<ul class="unordered-list"><li>Mappings between mouse variants in MoG+ and human variants in TogoVar are now available on gene pages (e.g., <a class="hyper-text -internal" href="/gene/1101#mouse-variants-in-mogplus-database">BRCA2</a>).</li><li>The mappings are based on the <a class="hyper-text -external" href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/">UCSC Chain File (hg38ToMm39.over.chain.gz)</a>.</li><li>You can now search for laboratory mouse strains that carry variants corresponding to human missense or pathogenic variants.</li><li>MoG+ is developed and operated by <a class="hyper-text -external" href="https://web.brc.riken.jp/en/">RIKEN BioResource Research Center</a>.</li></ul></li><li><strong>Gene page:</strong> Protein Browser was newly added to the gene pages.<ul class="unordered-list"><li>The following data are aligned along an amino acid sequence to facilitate comparison: (e.g., <a class="hyper-text -internal" href="/gene/1101#protein-browser">BRCA2</a>):<ul class="unordered-list"><li>Missense variants and their pathogenicity (Source: TogoVar)</li><li>Phosphorylation sites (Source: <a class="hyper-text -external" href="https://jpostdb.org/">jPOST: Japan ProteOme STandard Repository/Database</a>)</li><li>Glycosylation sites (Source: <a class="hyper-text -external" href="https://glycosmos.org/">GlyCosmos: Glycoscience Portal</a>)</li><li>Disease-associated variants (Source: UniProt)</li></ul></li></ul></li><li><strong>Variant page:</strong> The URL specification method was extended.<ul class="unordered-list"><li>By chromosome number, position, reference allele, and alternate allele:<ul class="unordered-list"><li>Example: <a class="hyper-text -internal" href="/variant/12-111803962-G-A">https://grch38.togovar.org/variant/12-111803962-G-A</a></li></ul></li><li>By dbSNP rs number:<ul class="unordered-list"><li>Example: <a class="hyper-text -internal" href="/variant/rs671">https://grch38.togovar.org/variant/rs671</a></li></ul></li></ul></li></ul><h3 class="sectiontitle">2024/12/3</h3><ul class="unordered-list"><li><strong>Datasets:</strong> Clinical significance from <a class="hyper-text -external" href="https://mgend.ncgm.go.jp/">MGeND (Medical Genomics Japan Variant Database)</a> was newly added.<ul class="unordered-list"><li>MGeND is a database of genome variants and their clinical characteristics, collected through <a class="hyper-text -external" href="https://www.amed.go.jp/en/program/list/14/05/006.html">the Clinical Genome Information Integration Database Development Project</a> by the Japan Agency for Medical Research and Development (AMED). MGeND contains variant information for diseases such as cancer, rare and intractable diseases, infectious diseases, dementia, and hearing loss, and can be considered the Japanese equivalent of ClinVar.</li><li>The disease names in MGeND have been aligned with the vocabulary defined in <a class="hyper-text -external" href="https://www.ncbi.nlm.nih.gov/medgen">MedGen</a>, which is also used in ClinVar, wherever possible.</li></ul></li><li><strong>Datasets:</strong> Added datasets derived from JGA and renamed some of them.<ul class="unordered-list"><li>We reanalyzed whole-genome sequences deposited in the <a class="hyper-text -external" href="https://humandbs.dbcls.jp/en/">NBDC Human Database</a>/<a class="hyper-text -external" href="https://www.ddbj.nig.ac.jp/jga/index-e.html">Japanese Genotype-phenotype Archive (JGA)</a>, aggregated allele frequencies and genotype counts, and released them as <a class="hyper-text -internal" href="/doc/datasets/jga_wgs">JGA-WGS</a>. The reference genome used is GRCh38. For more information on the reanalysis, please refer to <a class="hyper-text -external" href="https://humandbs.dbcls.jp/en/data-processing">“Whole genome sequencing analysis data (germline)”</a> on the NBDC Human Database website.</li><li>The frequencies previously published as JGA-NGS, obtained by variant calling on whole-exome sequences deposited to JGA, were renamed to <a class="hyper-text -internal" href="/doc/datasets/jga_wes">JGA-WES</a>.</li></ul></li><li><strong>Gene page:</strong> Missense variants are visualized on a 3D protein structure.<ul class="unordered-list"><li>The visualization helps you to assess the potential impact of each missense variant on a protein structure (e.g., <a class="hypertext -external" href="/gene/1097#protein-structure">BRAF:p.V600E</a>, <a class="hypertext -external" href="/gene/6407#protein-structure">KRAS:p.G12D</a>).</li><li>You can choose from protein structures registered in <a class="hyper-text -external" target="_blank" href="https://pdbj.org/?lang=en">Protein Data Bank Japan (PDBj)</a> or those predicted by <a class="hyper-text -external" target="_blank" href="https://alphafold.ebi.ac.uk/">AlphaFold</a>. Please note that the displayed structures do not account for the effects of the variant.</li></ul></li></ul><h3 class="sectiontitle">2024/6/27</h3><ul class="unordered-list"><li><strong>Datasets:</strong> Allele frequencies and genotype counts (by disease and sex) for <a class="hyper-text -internal" href="datasets/jga_snp#diseases_jgad000123">45 diseases</a> were added to <a class="hyper-text -internal" href="datasets/jga_snp">JGA-SNP</a> as disease-specific information.<ul class="unordered-list"><li><a class="hyper-text -external" href="https://biobankjp.org/en/">BioBank Japan</a>and <a class="hyper-text -external" href="https://www.ims.riken.jp/english/">RIKEN</a>collected samples and performed genotyping by SNP array for 182,557 individuals.</li><li>The frequency can be displayed by expanding the “Section: Frequency” &gt; “Dataset: "JGA-SNP” &gt; “Population: Total” on each variant report page (e.g., <a class="hyper-text -internal" href="/variant/tgv66359566#frequency">tgv66359566</a>). You can search for variants based on the disease and sex-specific frequencies using the “Alternative allele frequency/count” in the Advanced search.</li></ul></li><li><strong>Datasets:</strong> Allele frequencies and genotype counts of 9,290 Japanese were newly added as <a class="hyper-text -internal" href="datasets/ncbn">NCBN dataset</a>(GRCh38 only).<ul class="unordered-list"><li><a class="hyper-text -external" href="https://ncbiobank.org/en/">The National Center BioBank Network (NCBN)</a> collected the samples and conducted whole-genome analysis.</li><li>The Japanese frequencies can be compared with those of the 2,504 individuals from <a class="hyper-text -external" href="https://www.internationalgenome.org/">the international 1000 Genomes Project (1KGP)</a>. Joint calling was performed for the Japanese and 1KGP samples.</li><li>The frequency can be displayed by expanding the “Section: Frequency” &gt; “Dataset: NCBN” &gt; “Population: Total” on each variant report page (e.g., <a class="hyper-text -internal" href="/variant/tgv80918483#frequency">tgv80918483</a>). You can search for variants based on the population-specific frequencies using the “Alternative allele frequency/count” in the Advanced search.</li></ul></li></ul><h3 class="sectiontitle">2024/2/28</h3><ul class="unordered-list"><li><strong>Simple/Advanced search:</strong> Pathogenicity predictions with <a class="hyper-text -external" href="https://github.com/google-deepmind/alphamissense">AlphaMissense</a> (<a class="hyper-text -external" href="https://www.science.org/doi/10.1126/science.adg7492">Cheng et al.2023</a>) were added to the missense variants in TogoVar.</li><li><strong>Datasets:</strong> <a class="hyper-text -external" href="https://gnomad.broadinstitute.org/">The Genome Aggregation Database (gnomAD)</a> was updated to version 4.0.</li></ul><h3 class="sectiontitle">2023/11/16</h3><ul class="unordered-list"><li><strong>Datasets:</strong> ToMMo 14KJPN was updated to ToMMo 54KJPN (<a class="hyper-text -external" href="https://jmorp.megabank.tohoku.ac.jp/downloads/tommo-54kjpn-20230626-af_snvindelall">ToMMo 54KJPN-SNV/INDEL Allele Frequency Panel (v20230626)</a>).</li><li><strong>Advanced search:</strong> You can search by TogoVar ID (e.g. tgv47264307) or dbSNP refSNP ID (e.g. rs671).</li><li><strong>Downloads:</strong> Allele frequencies for all variants are available in <a class="hyper-text -external" href="https://samtools.github.io/hts-specs/VCFv4.4.pdf">VCF format</a> as well as tab-separated format.</li></ul><h3 class="sectiontitle">2023/7/14</h3><ul class="unordered-list"><li><strong>Simple/Advanced search:</strong> You can download search results in JSON, CSV, or TSV formats.</li><li><strong>Advanced search:</strong> We added the "Disease" field to the "Add condition" menu. You can now search by disease names from <a class="hyper-text -external" href="https://www.ncbi.nlm.nih.gov/medgen/">MedGen</a>. The search targets include diseases in the subconcepts of <a class="hyper-text -external" href="https://monarchinitiative.org/">the Mondo disease ontology</a>.</li></ul><h3 class="sectiontitle">2022/11/1</h3><ul class="unordered-list"><li><strong>URL:</strong> The URL of TogoVar was changed from <a class="hyper-text -external" href="https://togovar.biosciencedbc.jp">https://togovar.biosciencedbc.jp</a> to <a class="hyper-text -internal" href="https://togovar.org">https://togovar.org.</a></li><li><strong>Simple/Advanced search:</strong> You can search variants by a location of the GRCh38 reference sequence. Search by a GRCh37 location is still available.<ul class="unordererd-list"><li><a class="hyper-text -internal" href="https://grch38.togovar.org">https://grch38.togovar.org</a> (GRCh38)</li><li><a class="hyper-text -internal" href="https://grch37.togovar.org">https://grch37.togovar.org</a> (GRCh37)</li></ul></li><li><strong>Downloads:</strong> Bulk download of the GRCh38-based variant data was made available in addition to the GRCh37-based one.<ul class="unordererd-list"><li><a class="hyper-text -internal" href="https://grch38.togovar.org/downloads/release/current/grch38/">https://grch38.togovar.org/downloads/release/current/grch38/</a> (GRCh38)</li><li><a class="hyper-text -internal" href="https://grch37.togovar.org/downloads/release/current/grch37/">https://grch37.togovar.org/downloads/release/current/grch37/</a> (GRCh37)</li></ul></li><li><strong>Misc:</strong> The provider of TogoVar was changed from <a class="hyper-text -external" href="https://biosciencedbc.jp">Department of NBDC Program, Japan Science and Technology Agency</a>to <a class="hyper-text -external" href="https://dbcls.rois.ac.jp">Database Center for Life Science, Joint Support-Center for Data Science Research, Research Organization of Information and Systems (DBCLS).</a></li></ul><h3 class="sectiontitle">2022/8/5</h3><ul class="unordered-list"><li><strong>Advanced search:</strong> <a class="hyper-text -internal" href="/?mode=advanced">Advanced search</a>was released.</li><li><strong>Gene page/Disease page:</strong> Variant information pages related with a gene（e.g.：<a class="hyper-text -internal" href="/gene/8618">PAX4</a>) or a disease (e.g.：<a class="hyper-text -internal" href="/disease/C0023467">Acute myeloid leukemia </a>) was released.</li><li><strong>Variant page/Gene page/Disease page:</strong> Genome-wide association studies from <a class="hyper-text -external" href="https://www.ebi.ac.uk/gwas/">GWAS Catalog</a>was added to the variant, gene and disease pages.</li><li><strong>TogoVar API:</strong> <a class="hyper-text -internal" href="/api">TogoVar API version 0.91</a>was released.</li></ul><h3 class="sectiontitle">2021/12/16</h3><ul class="unordered-list"><li><strong>Datasets:</strong> ToMMo 4.7KJPN was updated to ToMMo 8.3KJPN (<a class="hyper-text -external" href="https://jmorp.megabank.tohoku.ac.jp/202109/downloads#variant">ToMMo 8.3KJPN Allele Frequency Panel).</a></li><li><strong>Datasets:</strong> ExAC was updated to gnomAD v2.1.1 (<a class="hyper-text -external" href="https://gnomad.broadinstitute.org/downloads">The Genome Aggregation Database).</a></li></ul><h3 class="sectiontitle">2020/10/22</h3><ul class="unordered-list"><li><strong>Keyword search:</strong> You can search TogoVar by <a class="hyper-text -external" target="_blank" href="https://varnomen.hgvs.org/">Human Genome Variation Society (HGVS) description</a>(e.g.<a target="_blank" href="/?term=ALDH2%3Ac.1510G%3EA">ALDH2:c.1510G&gt;A</a> or <a target="_blank" href="/?term=ALDH2%3Ac.1510G%3EA">ALDH2:p.Glu504Lys</a>).</li></ul><h3 class="sectiontitle">2020/7/27</h3><ul class="unordered-list"><li><strong>Datasets:</strong> The new dataset, <a href="/doc/datasets/gem_j_wga">GEM-Japan Whole Genome Aggregation (GEM-J WGA)</a>, was added. Additionally, ToMMo 3.5KJPN was updated to ToMMo 8.3KJPN (<a class="hyper-text -external" href="https://jmorp.megabank.tohoku.ac.jp/202001/downloads#variant">ToMMo 4.7KJPN Allele Frequency Panel</a>). ClinVar information was also updated.</li><li><strong>Usability:</strong> You can search the variants on Karyotype image.</li></ul><h3 class="sectiontitle">2019/7/25</h3><ul class="unordered-list"><li><strong>Keyword search:</strong> You can search TogoVar by an alias name of HGNC Gene Symbol (e.g. PD-1, p53).</li><li><strong>Filters:</strong> You can specify either all or any of datasets of your choice as a target when searching by an alternative allele frequency.</li><li><strong>Filters:</strong> You can filter variants by Consequence value, SIFT and/or Polyphen score.</li><li><strong>Filters:</strong> You can hide and show low quality variants.</li><li><strong>Datasets:</strong> ToMMo 3.5KJPN was updated to Ver.2 (<a class="hyper-text -external" href="https://jmorp.megabank.tohoku.ac.jp/201905/downloads/">ToMMo 3.5KJPNv2 Allele Frequency Panel</a>). ClinVar, Variant Effect Predictor (VEP) and Ensembl were updated as well. See <a href="/doc/datasets">Datasets</a> for detail.</li><li><strong>Usability:</strong> You can choose the data items to display at the Configuration menu.</li><li><strong>Usability:</strong> The user interface was refined for easier use.</li></ul><h3 class="sectiontitle">2018/6/7</h3><ul class="unordered-list"><li>TogoVar has been released. See <a class="hyper-text -external" href="https://www.eurekalert.org/pub_releases/2018-07/jsat-jr071018.php">a EurekAlert! press release</a> for detail.</li></ul></section></div></article></main></body></html>