<!doctype html><html data-page="datasets" lang="en"><head><title>Datasets | TogoVar</title><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><link rel="icon" href="/favicon.svg" type="image/svg+xml"><script>(async function() {
        try {
          if (window.location.origin === 'http://localhost:8000') return
          const response = await fetch(`${window.location.origin}/auth/status`);
      
          let isLogin = false;
          if (response.status === 200 || response.status === 403) {
            isLogin = true;
          }
      
          const loginMenu = document.querySelector('.menu-button.login a.link');
          if (isLogin === true) {
            loginMenu.href = "/auth/logout";
            loginMenu.textContent = "Logout";
          } else {
            loginMenu.href = "/auth/login";
            loginMenu.textContent = "Login";
          }
      
        } catch (error) {
          console.error('Error fetching auth status:', error);
        }
      })();</script><link href="/css/main-06f76ee01cc97734dbff.css" rel="stylesheet"></head><body><header id="GlobalHeader"><div class="left"><h1 class="heading"><a class="link" href="/"><img class="logotype" src="/images/logotype-9dd64508bb1747791269e1d69fc69389.svg" alt="TOGO VAR"></a></h1><span class="summary">A comprehensive Japanese genetic variation database</span><nav class="dropdown-view"><button class="button">GRCh38</button><div class="selectshow"><ul class="list"><li class="item"><a class="link" href="http://grch37.togovar.org/">GRCh37</a></li><li class="item"><a class="link -active">GRCh38</a></li></ul></div></nav></div><div class="right"><div class="menu-bar"></div><nav class="menus-view"><div class="menu-wrapper"><ul class="menu navigation"><li class="item menu-button"><a class="link" href="/">Home</a></li><li class="item menu-button -current"><a class="link" href="/doc/datasets">Datasets</a></li><li class="item menu-button"><a class="link" href="/doc/downloads">Downloads</a></li><li class="item menu-button"><a class="link" href="/api" target="_blank">API</a></li><li class="item menu-button"><a class="link" href="/doc/terms">Terms</a></li><li class="item menu-button"><a class="link" href="/doc/contact">Contact</a></li><li class="item menu-button"><a class="link" href="/doc/about">About</a></li><li class="item menu-button"><a class="link" href="/doc/history">History</a></li><li class="item menu-button"><a class="link" href="/doc/help">Help</a></li></ul><ul class="menu"><li class="menu-button languageswitcher en"><a class="language en" href="/doc/datasets">en</a><a class="language ja" href="/doc/ja/datasets">ja</a></li><li class="menu-button login"><a class="link"></a></li></ul></div></nav></div></header><main><article class="common-document"><header class="header"><h2 class="title">TogoVar datasets (GRCh38)</h2></header><div class="contents"><section class="section"><h3 class="sectiontitle">Variant frequencies for which you can apply for use of individual-level data<sup>∗1</sup> to the NBDC human databases<sup>∗2</sup></h3><p>Click the links at the Included controlled-access datasets to apply for use of individual-level data</p><table class="table"><thead><tr><th>Variant dataset name</th><th>Analysis method</th><th>Liftover GRCh37</th><th>Target population</th><th>Healthy subjects</th><th>Affected subjects</th><th>Sample size</th><th>Number of alleles</th><th>Included controlled-access datasets</th></tr></thead><tbody><tr><td><a href="/doc/datasets/gem_j_wga">GEM Japan Whole Genome Aggregation (GEM-J WGA) Panel</a></td><td class="_align-left">WGS</td><td class="_align-center">✔</td><td>Japanese</td><td class="_align-center">✔</td><td class="_align-center"></td><td class="_align-center"><span style="margin-left:35px;">7,609</span></td><td class="_align-center">95,555,740</td><td class="_align-center"><a href="/doc/datasets/gem_j_wga#jga_datasets">6 datasets</a></td></tr><tr><td><a href="/doc/datasets/jga_snp">JGA-SNP</a></td><td class="_align-left">SNP-Chip</td><td class="_align-center">✔</td><td>Japanese</td><td class="_align-center">✔</td><td class="_align-center">✔</td><td class="_align-center"><span style="margin-left:15px;">183,884</span></td><td class="_align-center"><span style="margin-left:10px;">1,249,671</span></td><td class="_align-center"><a href="/doc/datasets/jga_snp#jga_datasets">3 datasets</a></td></tr><tr><td><a href="/doc/datasets/jga_wes">JGA-WES</a></td><td class="_align-left">WES</td><td class="_align-center">✔</td><td>Japanese</td><td class="_align-center">✔</td><td class="_align-center">✔</td><td class="_align-center"><span style="margin-left:40px;">125</span></td><td class="_align-center"><span style="margin-left:10px;">4,668,805</span><br></td><td class="_align-center"><a href="/doc/datasets/jga_wes#jga_datasets">7 datasets</a></td></tr><tr><td><a href="/doc/datasets/jga_wgs">JGA-WGS</a></td><td class="_align-left">WGS</td><td class="_align-center"></td><td>Japanese</td><td class="_align-center">✔</td><td class="_align-center">✔</td><td class="_align-center"><span style="margin-left:45px;">78</span></td><td class="_align-center"><span style="margin-left:4px;">20,822,413</span></td><td class="_align-center"><a href="/doc/datasets/jga_wgs#jga_datasets">4 datasets</a></td></tr></tbody></table><p class="notice">∗1：fastq/bam/cel files and/or lists of genotype data etc.<br>∗2：<a class="hyper-text -external" target="_blank" href="https://www.ddbj.nig.ac.jp/jga/index.html">Japanese Genotype-phenotype Archive (JGA)</a> / <a class="hyper-text -external" target="_blank" href="https://gr-sharingdbs.dbcls.jp/agd">AMED Genome group sharing Database (AGD)</a></p></section><section class="section"><h3 class="sectiontitle">Other variant frequency datasets</h3><table class="table _column-span-all"><thead><tr><th>Variant dataset name</th><th>Analysis method</th><th>Liftover GRCh37</th><th>Target population</th><th>Healthy subjects</th><th>Affected subjects</th><th>Sample size</th><th>Number of alleles</th><th>Author</th><th>Version/Last updated</th></tr></thead><tbody><tr><td><a class="hyper-text -external" target="_blank" href="https://gnomad.broadinstitute.org/downloads#v4">Genome Aggregation Database (gnomAD) exomes</a></td><td class="_align-left">WES</td><td class="_align-center"></td><td class="_align-left">Mixed</td><td class="_align-center">✔</td><td class="_align-center">✔</td><td class="_align-center"><span style="margin-left:3px;">730,947</span></td><td class="_align-center">183,717,261</td><td>Broad Institute</td><td>v4.1</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://gnomad.broadinstitute.org/downloads#v4">Genome Aggregation Database (gnomAD) genomes</a></td><td class="_align-left">WGS</td><td class="_align-center"></td><td class="_align-left">Mixed</td><td class="_align-center">✔</td><td class="_align-center">✔</td><td class="_align-center"><span style="margin-left:10px;">76,215</span></td><td class="_align-center">759,336,320</td><td>Broad Institute</td><td>v4.1</td></tr><tr><td><a href="/doc/datasets/ncbn">NCBN</a></td><td class="_align-left">WGS</td><td class="_align-center">✔</td><td>Mixed</td><td class="_align-center">✔</td><td class="_align-center">✔</td><td class="_align-center"><span style="margin-left:9px;">JPN:9,290 1KGP:2,504</span></td><td class="_align-center">215,729,032</td><td>National Center Biobank Network (NCBN)</td><td>2024/6/28</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://jmorp.megabank.tohoku.ac.jp/downloads/tommo-54kjpn-20230626-af_snvindelall">ToMMo 54KJPN Allele Frequency Panel(54KJPN)</a></td><td class="_align-left">WGS</td><td class="_align-center"></td><td class="_align-left">Japanese</td><td class="_align-center">✔</td><td class="_align-center"></td><td class="_align-center"><span style="margin-left:10px;">54,302</span></td><td class="_align-center">252,777,334</td><td>Tohoku Medical Megabank Organization</td><td>v20230626</td></tr></tbody></table><p class="notice">Note: 54KJPN consists of SNVs (Autosome, chrX(PAR1+PAR2+XTR) and chrMT) and INDELs (Autosome and chrX(PAR1+PAR2+XTR)).</p></section><section class="section"><h3 class="sectiontitle">Non-variant frequency datasets</h3><table class="table"><thead><tr><th>Dataset name</th><th>Version/Last update</th><th>Description</th><th>Author</th></tr></thead><tbody><tr><td><a class="hyper-text -external" target="_blank" href="https://ftp.ncbi.nlm.nih.gov/pub/clinvar/">ClinVar</a></td><td>2024/10/03</td><td>Clinical significance of variants</td><td>NCBI</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://ftp.dbcls.jp/colil">Colil</a></td><td>Obtained by API</td><td>Information on citation relationships in life sciences literature</td><td>DBCLS</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://www.ncbi.nlm.nih.gov/grc/human/data?asm=GRCh38.p13">GRCh38.p13</a></td><td>2019/03/01</td><td>Human genome reference sequence</td><td>GRC</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://www.ebi.ac.uk/gwas/docs/file-downloads">GWAS Catalog</a></td><td>2024/10/15</td><td>Catalog of human genome-wide association studies</td><td>NHGRI-EBI</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://ftp.ebi.ac.uk/pub/databases/genenames/hgnc/tsv/hgnc_complete_set.txt">HGNC symbol report</a></td><td>2024/08/27</td><td>Approved human gene nomenclature and associated gene information</td><td>HGNC</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://www.ncbi.nlm.nih.gov/research/litvar2/api">LitVar</a></td><td>Obtained by API</td><td>Information on papers in which the names of variants appear</td><td>NCBI</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://www.ncbi.nlm.nih.gov/medgen">MedGen</a></td><td>2024/10/15</td><td>Information about conditions and phenotypes related to Medical Genetics</td><td>NCBI</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://mgend.ncgm.go.jp/">MGeND</a></td><td>2024/03/08</td><td>Clinical significance of variants collected from the Japanese population</td><td>NCGM</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://ftp.ncbi.nlm.nih.gov/pubmed/">PubMed</a></td><td>2024/10/14</td><td>Information on papers</td><td>NCBI</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://ftp.ncbi.nlm.nih.gov/pub/lu/PubTatorCentral/">PubTator Central</a></td><td>2024/01/04</td><td>Information on papers in which the names of variants appear</td><td>NCBI</td></tr></tbody></table><p class="notice">Note: TogoVar obtained ClinVar variants from <a class="hyper-text -external" target="_blank" href="https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/">the VCF file</a>that contains only variants for which GRCh38 positions were determined.<br>Note: Disease names in MGeND were mapped to MedGen using a proprietary method, so they may not match the disease names listed in MGeND.</p></section><section class="section"><h3 class="sectiontitle">Tools for data processing</h3><table class="table"><thead><tr><th>Name</th><th>Ver.</th><th>Description</th><th>Author</th></tr></thead><tbody><tr><td><a class="hyper-text -external" target="_blank" href="https://samtools.github.io/bcftools/">bcftools</a></td><td>1.20</td><td>Split multiallelic sites into biallelic variants, exclude reference mismatches, and normalize them</td><td>Genome Research Ltd.</td></tr><tr><td><a href="/doc/datasets/analysis">BioReT</a></td><td>‐</td><td>Identify germline short variants (SNPs and Indels) in WES from JGA to produce a joint callset in VCF format</td><td><a class="hyper-text -external" target="_blank" href="https://amelieff.jp/">Amelieff</a></td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://gatk.broadinstitute.org/hc/en-us/articles/360035535932-Germline-short-variant-discovery-SNPs-Indels">GATK Best Practice - Germline short variant discovery (SNPs and Indels)</a></td><td>4</td><td>Identify germline short variants (SNPs and Indels) in WGS from JGA to produce a joint callset in VCF format.</td><td>Broad Institute</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://github.com/informationsea/transanno">transanno</a></td><td>0.45</td><td>Accurate VCF LiftOver tool for converting genome assemblies from GRCh37 to GRCh38</td><td>OKAMURA, Yasunobu</td></tr><tr><td><a class="hyper-text -external" target="_blank" href="https://www.ensembl.org/info/docs/tools/vep/index.html">Variant Effect Predictor (VEP)</a></td><td>Ensembl release 112</td><td>Add <a class="hyper-text -external" target="_blank" href="https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#cache_content">Annotations like gene names, consequences or deleterious predictions (AlphaMissense, SIFT and Polyphen)</a> to variants</td><td>EMBL-EBI</td></tr></tbody></table></section></div></article></main></body></html>