<!doctype html><html data-page="help" lang="en"><head><title>Help | TogoVar</title><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><link rel="icon" href="/favicon.svg" type="image/svg+xml"><script>(async function() {
        try {
          if (window.location.origin === 'http://localhost:8000') return
          const response = await fetch(`${window.location.origin}/auth/status`);
      
          let isLogin = false;
          if (response.status === 200 || response.status === 403) {
            isLogin = true;
          }
      
          const loginMenu = document.querySelector('.menu-button.login a.link');
          if (isLogin === true) {
            loginMenu.href = "/auth/logout";
            loginMenu.textContent = "Logout";
          } else {
            loginMenu.href = "/auth/login";
            loginMenu.textContent = "Login";
          }
      
        } catch (error) {
          console.error('Error fetching auth status:', error);
        }
      })();</script><link href="/css/main-06f76ee01cc97734dbff.css" rel="stylesheet"><script src="/js/main-68621748ac045a215be3.js"></script></head><body><header id="GlobalHeader"><div class="left"><h1 class="heading"><a class="link" href="/"><img class="logotype" src="/images/logotype-9dd64508bb1747791269e1d69fc69389.svg" alt="TOGO VAR"></a></h1><span class="summary">A comprehensive Japanese genetic variation database</span><nav class="dropdown-view"><button class="button">GRCh37</button><div class="selectshow"><ul class="list"><li class="item"><a class="link -active">GRCh37</a></li><li class="item"><a class="link" href="http://grch38.togovar.org/">GRCh38</a></li></ul></div></nav></div><div class="right"><div class="menu-bar"></div><nav class="menus-view"><div class="menu-wrapper"><ul class="menu navigation"><li class="item menu-button"><a class="link" href="/">Home</a></li><li class="item menu-button"><a class="link" href="/doc/datasets">Datasets</a></li><li class="item menu-button"><a class="link" href="/doc/downloads">Downloads</a></li><li class="item menu-button"><a class="link" href="/api" target="_blank">API</a></li><li class="item menu-button"><a class="link" href="/doc/terms">Terms</a></li><li class="item menu-button"><a class="link" href="/doc/contact">Contact</a></li><li class="item menu-button"><a class="link" href="/doc/about">About</a></li><li class="item menu-button"><a class="link" href="/doc/history">History</a></li><li class="item menu-button -current"><a class="link" href="/doc/help">Help</a></li></ul><ul class="menu"><li class="menu-button languageswitcher en"><a class="language en" href="/doc/help">en</a><a class="language ja" href="/doc/ja/help">ja</a></li><li class="menu-button login"><a class="link"></a></li></ul></div></nav></div></header><main><article class="common-document"><header class="header"><h2 class="title">Help</h2></header><div class="contents"><section class="section"><ul class="page-index"><li class="category"><h3 class="categorytitle">TogoVar in general</h3><ul class="categorylist"><li class="item -question"><a class="link" href="#g1">What are the objectives and utilities of TogoVar?</a></li><li class="item -question"><a class="link" href="#g2">Do you have videos that explain TogoVar?</a></li></ul></li><li class="category"><h3 class="categorytitle">About the terms of use</h3><ul class="categorylist"><li class="item -question"><a class="link" href="#t1">Can I use the TogoVar for free?</a></li><li class="item -question"><a class="link" href="#t2">How to cite the TogoVar database in my publication?</a></li><li class="item -question"><a class="link" href="#t3">Can I use it for commercial purposes?</a></li></ul></li><li class="category"><h3 class="categorytitle">About the data collected in TogoVar</h3><ul class="categorylist"><li class="item -question"><a class="link" href="#d1">Can I retrieve phenotype information on an individual level?</a></li><li class="item -question"><a class="link" href="#d2">Are the data linked with information about gene expression and proteins?</a></li><li class="item -question"><a class="link" href="#d3">What software is used for detecting variants and how you manage the quality?</a></li><li class="item -question"><a class="link" href="#d4">What are the targeted variant types for TogoVar?</a></li><li class="item -question"><a class="link" href="#d5">How do you calculate the Consequence value?</a></li><li class="item -question"><a class="link" href="#d6">How do you calculate the value of Clinical significance?</a></li><li class="item -question"><a class="link" href="#d7">What are AlphaMissense, SIFT and PolyPhen?</a></li><li class="item -question"><a class="link" href="#d8">Was a liftover tool used for generating variant data?</a></li></ul></li><li class="category"><h3 class="categorytitle">About the TogoVar search system</h3><ul class="categorylist"><li class="item -question"><a class="link" href="#s1">What does the frequency indicator in the Frequency column of the search results represent?</a></li><li class="item -question"><a class="link" href="#s2">For each variant type, how is the position, reference and alternative allele described respectively?</a></li><li class="item -question"><a class="link" href="#s3">In TogoVar, how is the variant described?</a></li><li class="item -question"><a class="link" href="#s4">Can I search by HGVS description (e.g. ALDH2:p.Glu504Lys)?</a></li><li class="item -question"><a class="link" href="#s5">What can I do with the Advanced search?</a></li><li class="item -question"><a class="link" href="#s6">Can I search for variants with different alternative allele frequencies between Japanese and other populations?</a></li><li class="item -question"><a class="link" href="#s7">Can I search for variants by both positions on GRCh38 and GRCh37?</a></li></ul></li></ul></section></div><div class="contents"><section class="section"><h3 class="sectiontitle">TogoVar in general</h3><div class="question-answer" id="g1"><h4 class="question">What are the objectives and utilities of TogoVar?</h4><div class="answer"><p>NGS technologies have enabled large-scale genome sequencing to capture variants at low frequencies. However, existing databases do not contain sufficient variant frequency data for Japanese populations. To address this problem, TogoVar seeks to provide a reliable frequency information by assembling variant information from the NBDC Human Database/Japanese Genotype-phenotype Archive (JGA) and public Japanese genomic variation data.</p><p>In addition, since the reference human genome sequence can be updated, the genomic position may vary for the same variant if it refers to a different genome build. Thus, in our database, a unique ID is assigned to the same variant, thereby preventing the need to convert the positional information of the variant among different reference sequences, as done by researchers.</p></div></div><div class="question-answer" id="g2"><h4 class="question">Do you have videos that explain TogoVar?</h4><div class="answer"><p>See <a href="https://togotv.dbcls.jp/en/">TogoTV</a>. Videos on TogoVar are <a href="https://togotv.dbcls.jp/en/result.html?type=manual&amp;lang=en&amp;page=1&amp;query=TogoVar">here</a>.</p></div></div></section><section class="section"><h3 class="sectiontitle">About the terms of use</h3><div class="question-answer" id="t1"><h4 class="question">Can I use the TogoVar for free?</h4><div class="answer"><p>You can use the TogoVar service for free. For the use of the database, refer to the <a href="/doc/terms">Terms</a>.</p></div></div><div class="question-answer" id="t2"><h4 class="question">How to cite the TogoVar database in my publication?</h4><div class="answer"><p>Refer to the attribution (credit) display examples in <a href="/doc/terms">Terms</a>.</p></div></div><div class="question-answer" id="t3"><h4 class="question">Can I use it for commercial purposes?</h4><div class="answer"><p>You are free to browse the TogoVar websites and use the docker images to be released for commercial purposes. For data created by third parties and imported in TogoVar, please follow the terms of use of each organization. For more details, refer to <a href="/doc/terms">Terms</a>.</p></div></div></section><section class="section"><h3 class="sectiontitle">About the data collected in TogoVar</h3><div class="question-answer" id="d1"><h4 class="question">Can I retrieve phenotype information on an individual level?</h4><div class="answer"><p>TogoVar database does not keep private information, and we do not currently provide phenotype information for each individual. For more details about phenotype information, please directly contact the research groups described in the <a href="./datasets/jga_wes">JGA-WES dataset</a> , and <a href="/doc/datasets/jga_snp">JGA-SNP dataset</a>.</p></div></div><div class="question-answer" id="d2"><h4 class="question">Are the data linked with information about gene expression and proteins?</h4><div class="answer"><p>Not yet but planned.</p></div></div><div class="question-answer" id="d3"><h4 class="question">What software is used for detecting variants and how you manage the quality?</h4><div class="answer"><p>Refer to <a href="/doc/datasets/analysis">How JGA dataset was generated</a>.</p></div></div><div class="question-answer" id="d4"><h4 class="question">What are the targeted variant types for TogoVar?</h4><div class="answer"><p>There are five targeted variant types for TogoVar: SNV, Insertion, Deletion, Indel, and Substitution. Of <a class="hyper-text -external" target="_blank" href="http://varnomen.hgvs.org/bg-material/simple/#types-of-variants">5 types that HGVS enumerated as the types of variants</a>, Duplication is classified as Insertion, and Indels, for which the lengths of Insertion and Deletion are the same, are classified as Substitution. To identify the type of the variant, <a class="hyper-text -external" target="_blank" href="https://www.ensembl.org/vep">Variant Effect Predictor (VEP)</a> is used.</p></div></div><div class="question-answer" id="d5"><h4 class="question">How do you calculate the Consequence value?</h4><div class="answer"><p>The value of variant consequence in <a class="hyper-text -external" target="_blank" href="https://asia.ensembl.org/info/genome/variation/prediction/predicted_data.html#consequences">Variant Effect Predictor (VEP)</a> is displayed. If there are multiple transcripts for a single variant, only the most critical consequence may be displayed.</p></div></div><div class="question-answer" id="d6"><h4 class="question">How do you calculate the value of Clinical significance?</h4><div class="answer"><p>The value of <a class="hyper-text -external" target="_blank" href="https://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/#clinsig_options_scv">clinical significance in ClinVar</a> and <a class="hyper-text -external" target="_blank" href="https://mgend.ncgm.go.jp/about?lang=en#interpretation-link">clinical significance in MGeND</a> are displayed.</p></div></div><div class="question-answer" id="d7"><h4 class="question">What are AlphaMissense, SIFT and Polyphen?</h4><div class="answer"><p>AlphaMissense, SIFT and PolyPhen predict the impact of variants on protein function when an amino acid sequence is altered. These programs were performed using <a class="hyper-text -external" target="_blank" href="https://asia.ensembl.org/info/genome/variation/prediction/protein_function.html">Variant Effect Predictor (VEP)</a>. The icons below represent the threshold scores used for qualitative predictions.</p><ul class="unordered-list _margin-bottom _align-left"><li><strong>AlphaMissense</strong><br><div><div class="variant-function _width_5em _align-center" data-function="LP">&gt; 0.564</div><span class="_margin-left">Likely pathogenic</span></div><div><div class="variant-function _width_5em _align-center" data-function="A">&ge; 0.340</div><span class="_margin-left">Ambiguous</span></div><div><div class="variant-function _width_5em _align-center" data-function="LB">&lt; 0.340</div><span class="_margin-left">Likely benign</span></div><span>The score thresholds for "Likely pathogenic" and "Likely benign" were chosen to classify pathogenic and benign ClinVar variants with 90% precision, respectively. See </span><a class="hyper-text -external" target="_blank" href="https://asia.ensembl.org/info/docs/tools/vep/script/vep_plugins.html#alphamissense">AlphaMissense in VEP Plugins</a> for details.</li><li><strong>SIFT</strong><br><div><div class="variant-function _width_5em _align-center" data-function="D">&lt; 0.05</div><span class="_margin-left">Deleterious</span></div><div><div class="variant-function _width_5em _align-center" data-function="T">&ge; 0.05</div><span class="_margin-left">Tolerated</span></div><span>The score represents the normalized probability of an amino acid change being tolerated. Scores closer to zero indicate a higher likelihood of the change being deleterious. The qualitative prediction is based on this score. Substitutions with a score less than 0.05 are classified as "Deleterious", while all others are considered "Tolerated". See </span><a class="hyper-text -external" target="_blank" href="https://asia.ensembl.org/info/genome/variation/prediction/protein_function.html#sift">SIFT in Pathogenicity prediction of Ensembl Variation</a> for details.</li><li><strong>PolyPhen</strong><br><div><div class="variant-function _width_5em _align-center" data-function="PROBD">&gt; 0.908</div><span class="_margin-left">Probably Damaging</span></div><div><div class="variant-function _width_5em _align-center" data-function="POSSD">&gt; 0.446</div><span class="_margin-left">Possibly Damaging</span></div><div><div class="variant-function _width_5em _align-center" data-function="B">&le; 0.446</div><span class="_margin-left">Benign</span></div><div><div class="variant-function _width_5em _align-center" data-function="U">Unknown</div><span class="_margin-left">Unknown</span></div><span>Variants are predicted as "Probably Damaging" if they have posterior probability scores indicating an estimated false positive rate (FPR) of 5% or below. Those with FPRs at or below 10% are "Possibly Damaging", while variants with FPRs above 10% are classified as "Benign". See </span><a class="hyper-text -external" target="_blank" href="https://asia.ensembl.org/info/genome/variation/prediction/protein_function.html#polyphen">Polyphen in Pathogenicity prediction of Ensembl Variation</a> for details.</li></ul></div></div><div class="question-answer" id="d8"><h4 class="question">Was a liftover tool used for generating variant data?</h4><div class="answer"><p>Yes. Some of the GRCh38 datasets have been created by lifting over their GRCh37 datasets. Please refer to<a class="hyper-text -external" target="_blank" href="https://grch38.togovar.org/doc/datasets"> the Liftover GRCh37 column in the TogoVar datasets (GRCh38) page</a>for the target datasets. A liftover tool <a class="hyper-text -external" target="_blank" href="https://github.com/informationsea/transanno">transanno</a>has been used since the release on November 16, 2023. <a class="hyper-text -external" target="_blank" href="https://crossmap.sourceforge.net/">CrossMap</a>was used until the release on July 14, 2023.</p></div></div></section><section class="section"><h3 class="sectiontitle">About the TogoVar search system</h3><div class="question-answer _column-span-all" id="s1"><h4 class="question">What does the frequency indicator in the Frequency column of the search results represent?</h4><div class="answer"><p>The frequency indicator represents the alternative allele frequency for each dataset. The color of the frequency meter corresponds to the color allocated to the dataset. The data availability or frequency value is represented on a 9-point scale.</p><div class="_flex _break-avoid"><div class="_padding-right-large"><logarithmized-block-graph-frequency-view data-dataset="" data-logarithmized-frequency="na"></logarithmized-block-graph-frequency-view>&nbsp; Not Available<br><logarithmized-block-graph-frequency-view data-dataset="" data-logarithmized-frequency="monomorphic"></logarithmized-block-graph-frequency-view>&nbsp; Monomorphic (REF)<br><logarithmized-block-graph-frequency-view data-dataset="" data-logarithmized-frequency="undefined" data-count="1"></logarithmized-block-graph-frequency-view>&nbsp; Singleton</div><div class="_padding-right-large"><logarithmized-block-graph-frequency-view data-dataset="" data-logarithmized-frequency="&lt;0.0001"></logarithmized-block-graph-frequency-view>&nbsp; &lt; 0.0001<br><logarithmized-block-graph-frequency-view data-dataset="" data-logarithmized-frequency="&lt;0.001"></logarithmized-block-graph-frequency-view>&nbsp; &lt; 0.001<br><logarithmized-block-graph-frequency-view data-dataset="" data-logarithmized-frequency="&lt;0.01"></logarithmized-block-graph-frequency-view>&nbsp; &lt; 0.01<br><logarithmized-block-graph-frequency-view data-dataset="" data-logarithmized-frequency="&lt;0.05"></logarithmized-block-graph-frequency-view>&nbsp; &lt; 0.05<br><logarithmized-block-graph-frequency-view data-dataset="" data-logarithmized-frequency="&lt;0.5"></logarithmized-block-graph-frequency-view>&nbsp; &lt; 0.5<br><logarithmized-block-graph-frequency-view data-dataset="" data-logarithmized-frequency="≥0.5"></logarithmized-block-graph-frequency-view>&nbsp; ≥ 0.5</div><div class="_padding-right-large"><logarithmized-block-graph-frequency-view data-dataset="gem_j_wga" data-logarithmized-frequency="≥0.5"></logarithmized-block-graph-frequency-view>&nbsp; GEM-J WGA<br><logarithmized-block-graph-frequency-view data-dataset="jga_snp" data-logarithmized-frequency="≥0.5"></logarithmized-block-graph-frequency-view>&nbsp; JGA-SNP<br><logarithmized-block-graph-frequency-view data-dataset="jga_wes" data-logarithmized-frequency="≥0.5"></logarithmized-block-graph-frequency-view>&nbsp; JGA-WES<br><logarithmized-block-graph-frequency-view data-dataset="tommo" data-logarithmized-frequency="≥0.5"></logarithmized-block-graph-frequency-view>&nbsp; ToMMo 8.3KJPN<br><logarithmized-block-graph-frequency-view data-dataset="hgvd" data-logarithmized-frequency="≥0.5"></logarithmized-block-graph-frequency-view>&nbsp; HGVD<br><logarithmized-block-graph-frequency-view data-dataset="gnomad_genomes" data-logarithmized-frequency="≥0.5"></logarithmized-block-graph-frequency-view>&nbsp; gnomAD genomes<br><logarithmized-block-graph-frequency-view data-dataset="gnomad_exomes" data-logarithmized-frequency="≥0.5"></logarithmized-block-graph-frequency-view>&nbsp; gnomAD exomes</div></div></div></div><div class="question-answer _column-span-all" id="s2"><h4 class="question">For each variant type, how is the position, reference and alternative allele described respectively?</h4><div class="answer"><p>An example for each variant type is shown below. Note that the position of the first base in the chromosome is 1.</p><table class="results-view"><thead><tr><th><p>Variant Type</p></th><th><p>ID</p></th><th><p>Position</p></th><th><p>Ref / Alt</p></th><th><p>Description</p></th></tr></thead><tbody><tr><td><div class="variant-type">SNV</div></td><td><a target="_blank" href="/variant/tgv167913215">tgv167913215</a></td><td><div class="chromosome-position"><div class="chromosome">12</div><div class="coordinate">111767141</div></div></td><td><div class="ref-alt"><span class="ref">C</span><span class="arrow"></span><span class="alt">A</span></div></td><td>The 111767141st base C in chromosome 12 was replaced by A.</td></tr><tr><td><div class="variant-type">Ins</div></td><td><a target="_blank" href="/variant/tgv134135934">tgv134135934</a></td><td><div class="chromosome-position"><div class="chromosome">12</div><div class="coordinate">111786238</div></div></td><td><div class="ref-alt"><span class="ref"></span><span class="arrow"></span><span class="alt">G</span></div></td><td>A base G was inserted between the 111786238th and 111786239th bases in chromosome 12.</td></tr><tr><td><div class="variant-type">Del</div></td><td><a target="_blank" href="/variant/tgv110307325">tgv110307325</a></td><td><div class="chromosome-position"><div class="chromosome">12</div><div class="coordinate">111778836</div></div></td><td><div class="ref-alt"><span class="ref">A</span><span class="arrow"></span><span class="alt"></span></div></td><td>The 111778836th base A in Chromosome 1 was deleted.</td></tr></tbody></table></div></div><div class="question-answer _column-span-all" id="s3"><h4 class="question">In TogoVar, how is the variant described?</h4><div class="answer"><p>A. In TogoVar a variant is identified as a set of its genomic position, reference and alternative allele and human genome build. Examples of variant notations in TogoVar, dbSNP, <a class="hyper-text -external" target="_blank" href="https://samtools.github.io/hts-specs/VCFv4.2.pdf">VCF</a> and <a class="hyper-text -external" target="_blank" href="http://varnomen.hgvs.org/bg-material/simple/#types-of-variants">HGVS</a> are shown below.</p><table class="results-view"><thead><tr class="groupingrow"><th colspan="2"></th><th colspan="4"><p>TogoVar</p></th><th colspan="4"><p>dbSNP</p></th><th colspan="4"><p>VCF</p></th><th><p>HGVS</p></th></tr><tr><th></th><th class="separator"></th><th><p>ID</p></th><th><p>Position (GRCh38)</p></th><th><p>Ref / Alt</p></th><th class="separator"></th><th><p>ID</p></th><th><p>Position (GRCh38)</p></th><th><p>Alleles</p></th><th class="separator"></th><th><p>Position</p></th><th><p>Ref</p></th><th><p>Alt</p></th><th class="separator"></th><th></th></tr></thead><tbody><tr><td rowspan="2"><div class="variant-type">SNV</div></td><td class="separator" rowspan="2"></td><td><a target="_blank" href="/variant/tgv167913215">tgv167913215</a></td><td rowspan="2"><div class="chromosome-position"><div class="chromosome">12</div><div class="coordinate">111767141</div></div></td><td><div class="ref-alt"><span class="ref">C</span><span class="arrow"></span><span class="alt">A</span></div></td><td class="separator" rowspan="2"></td><td rowspan="2"><a class="hyper-text -external" target="_blank" href="https://www.ncbi.nlm.nih.gov/snp/rs567143183">rs567143183</a></td><td rowspan="2">chr12:111767141</td><td>C&gt;A</td><td class="separator" rowspan="2"></td><td rowspan="2">12:111767141</td><td>C</td><td>A</td><td class="separator" rowspan="2"></td><td>12:g.111767141:C&gt;A</td></tr><tr><td><a target="_blank" href="/variant/tgv47263142">tgv47263142</a></td><td><div class="ref-alt"><span class="ref">C</span><span class="arrow"></span><span class="alt">G</span></div></td><td>C&gt;G</td><td>C</td><td>G</td><td>12:g.111767141:C&gt;G</td></tr><tr><td><div class="variant-type">Ins</div></td><td class="separator"></td><td><a target="_blank" href="/variant/tgv134135934">tgv134135934</a></td><td><div class="chromosome-position"><div class="chromosome">12</div><div class="coordinate">111786238</div></div></td><td><div class="ref-alt"><span class="ref"></span><span class="arrow"></span><span class="alt">G</span></div></td><td class="separator"></td><td><a class="hyper-text -external" target="_blank" href="https://www.ncbi.nlm.nih.gov/snp/rs534925026">rs534925026</a></td><td>chr12:111786238</td><td>insG</td><td class="separator"></td><td>12:111786238</td><td>T</td><td>TG</td><td class="separator"></td><td>12:g.111786238_111786239insG</td></tr><tr><td><div class="variant-type">Del</div></td><td class="separator"></td><td><a target="_blank" href="/variant/tgv110307325">tgv110307325</a></td><td><div class="chromosome-position"><div class="chromosome">12</div><div class="coordinate">111778836</div></div></td><td><div class="ref-alt"><span class="ref">A</span><span class="arrow"></span><span class="alt"></span></div></td><td class="separator"></td><td><a class="hyper-text -external" target="_blank" href="https://www.ncbi.nlm.nih.gov/snp/rs766411401">rs766411401</a></td><td>chr12:111778836-111778847</td><td>delA</td><td class="separator"></td><td>12:111778835</td><td>GA</td><td>G</td><td class="separator"></td><td>12:g.111778847del</td></tr></tbody></table></div></div><div class="question-answer _column-span-all" id="s4"><h4 class="question">Q: Can I search by<a class="hyper-text -external" target="_blank" href="https://varnomen.hgvs.org/">HGVS description</a>(e.g. ALDH2:p.Glu504Lys)?</h4><div class="answer"><p>Yes. HGVSc (e.g.<a target="_blank" href="/?term=ALDH2%3Ac.1510G%3EA">ALDH2:c.1510G&gt;A</a>) and HGVSp (e.g.<a target="_blank" href="/?term=ALDH2%3Ap.Glu504Lys">ALDH2:p.Glu504Lys</a>) are acceptable. <a class="hyper-text -external" target="_blank" href="https://m.ensembl.org/info/docs/tools/vep/recoder/index.html">Variant Recoder</a>is used to convert an HGVS description to a GRCh38/GRCh37 position.</p></div></div><div class="question-answer _column-span-all" id="s5"><h4 class="question">Q: What can I do with the <a class="hyper-text -external" target="_blank" href="/?mode=advanced">Advanced search</a>?</h4><div class="answer"><p>You can build a query with an alternative allele frequency, clinical significance, and other search criteria combined with the AND and/or OR operator. See<a class="link" href="#s6">"Q: Can I search for variants with different alternative allele frequencies between Japanese and other populations?"</a> for usage.</p></div></div><div class="question-answer _column-span-all" id="s6"><h4 class="question">Can I search for variants with different alternative allele frequencies between Japanese and other populations?</h4><div class="answer"><p>Yes. You can search by<a class="hyper-text -external" target="_blank" href="/?mode=advanced"> Advanced search</a></p><p></p>Variant sets for comparison of alternative allele frequencies are as follows:<ul class="unordered-list"><li>For Japanese populations, variants included in either GEM-J WGA or 8.3KJPN</li><li>For gnomAD populations, variants included in either gnomAD genomes or gnomAD exomes</li></ul><p></p>We will search for variants with an alternative allele frequency greater than 0.3 for Japanese populations and less than 0.05 for gnomAD populations.<ol class="ordered-list"><li>Select the Advanced search tab at the top of<a class="hyper-text -external" target="_blank" href="https://togovar-stg.biosciencedbc.jp"> the home page.</a><img src="/images/help-image1-ddc98855f2d3382275a08c7f4af30505.png" alt="help-image1"></li><li>Select the Alternative allele frequency/count at the Add condition menu.<p><img src="/images/help-image2-7a298460c582b5832b23ad275ef394a9.png" alt="help-image2" width="241"></p></li><li>Specify a dataset (GEM-J WGA) and an alternative allele frequency (>= 0.3). You can also specify the number of alternative alleles.<img src="/images/help-image3-6f3b4020d18d40d07589fa590d06bc34.png" alt="help-image3"></li><li>Click the OK button. The search criteria has been set.<img src="/images/help-image4-4a5df54fbad1709bd8d71ed40a1f6964.png" alt="help-image4"></li><li>In the same procedures, enter search criteria for the datasets, ToMMo 8.3KJPN, gnomAD genomes and gnomAD exomes.<img src="/images/help-image5-20b7cbccbd7fe131a5c42898569aa66b.png" alt="help-image5"></li><li>Join the search criteria for Japanese populations, GEM-J WGA and ToMMo 8.3KJPN, with the OR operator.<ol class="ordered-list"><li>Select the search criteria for GEM-J WGA and ToMMo 8.3KJPN. The selected criteria is enclosed by yellow borders.<img src="/images/help-image6-3c4c541d713e23f409a5ee0de1ce91ac.png" alt="help-image6"></li><li>Select the Group menu. The selected criteria are joined with the OR operator. To change to the AND operator, click the OR icon.<img src="/images/help-image7-cfda02e6dcd665995fae3d49c960ff4f.png" alt="help-image7"></li></ol></li><li>In the same procedures, the search criteria for the gnomAD genomes and gnomAD exomes are joined with the OR operator.<img src="/images/help-image8-0085273d956212ad3e97d707d3e842e1.png" alt="help-image8"></li></ol><p></p>The final search criteria is as follows:<p></p>((freq(GEM-J WGA) &gt;= 0.3) OR (freq(ToMMo 8.3KJPN) &gt;= 0.3))<br>AND<br>(freq(gnomAD genomes) &lt;= 0.05) OR (freq(gnomAD exomes) &lt;= 0.05))<p></p>Each time search criteria is changed, the search is automatically executed and a list of variants that match the criteria is refreshed.<img src="/images/help-image9-8f9c07b7626f8c429d5ce28efe9a9b8f.png" alt="help-image9"><p></p>Functions in the menu bar<ul class="unordered-list"><li>Add condition: Add search criteria to the query builder.</li><li>Group: Combine selected search criteria indicated by yellow borders with the OR operators. Note: click the OR icon to change to the AND operator.</li><li>Ungroup: Separate joined search criteria. This is a reverse operation of the Group menu.</li><li>Delete: Delete selected search criteria indicated by yellow borders.</li></ul><p></p>Meaning of the icons<ul class="unordered-list"><li><img src="/images/help-pencil-2c996abde858a1595bafe2fd46316dab.png" alt="help-pencil">：Edit search criteria</li><li><img src="/images/help-cross-034f5af3c41ab0b98341c9faf2559713.png" alt="help-cross">：Delete search criteria</li></ul><p></p></div></div><div class="question-answer _column-span-all" id="s7"><h4 class="question">Can I search for variants by both positions on GRCh38 and GRCh37?</h4><div class="answer"><p></p>Yes. The version number of the reference sequence is displayed at the header. Click the version number to switch to the other.<img src="/images/help-image10-ca1f7031cfd252accc478d0c7c6593f7.png" alt="help-image10"></div></div></section></div></article></main></body></html>